api/src/utils/put-utils.js

```diff
@@ -112,3 +114,71 @@ exports.getDotNotation = function(obj, target, prefix) {
 
   return target;
 };
+
+/**
+ * Updates the read roles for all documents associated with the master record.
+ *
+ * @param {*} masterRecord
+ * @param {*} auth_payload
+ * @returns savedDocuments
+ */
+exports.updateDocumentRoles = async function(masterRecord, auth_payload) {
+  if (!masterRecord) {
+    return null;
+  }
+
+  let savedDocuments = [];
+  const documentPromises = [];
+
+  if (await QueryUtils.isDocumentConsideredAnonymous(masterRecord)) {
+    // unpublish the mongo document AND s3 document if one exists
+    masterRecord.documents.forEach(docId => {
+      documentPromises.push(
+        DocumentController.unpublishDocument(docId, auth_payload).then(document => {
+          if (document.key) {
+            return DocumentController.unpublishS3Document(document.key);
+          }
+        })
+      );
+    });
+  } else {
+    // publish the mongo document AND s3 document if one exists
+    masterRecord.documents.forEach(docId => {
+      documentPromises.push(
+        DocumentController.publishDocument(docId, auth_payload).then(document => {
+          if (document.key) {
+            return DocumentController.publishS3Document(document.key);
+          }
+        })
+      );
+    });
+  }
+
+  if (documentPromises.length > 0) {
+    savedDocuments = await Promise.all(documentPromises);
+  }
+
+  return savedDocuments;
+};
+
+/**
+ * Apply business logic changes to a record. Updates the provided updateObj, and returns it.
+ *
+ * @param {*} updateObj
+ * @param {*} sanitizedObj
+ * @returns updateObj
+ */
+exports.applyBusinessLogic = function(updateObj, sanitizedObj) {
+  if (!sanitizedObj) {
+    return updateObj;
+  }
+
+  // apply anonymous business logic
+  if (QueryUtils.isRecordConsideredAnonymous(sanitizedObj)) {
+    updateObj.$pull['issuedTo.read'] = 'public';
+  } else {
+    updateObj.$addToSet['issuedTo.read'] = 'public';
+  }
```

Comment:
Maybe passing in the object property instead of hardcoding here would be better:

`updateObj.$pull['issuedTo.read'] = 'public'; ` => `updateObj.$pull[someProperty] = 'public';`

I don't know if QueryUtils is where the real business logic should sit.  I personally like to see the consideration for the reading/modification of the roles, and determining anonymous/not anonymous in the same file.  Not spread out in put-utils, and not in query-utils, but in something like businessManager.js.  This way all the business logic is in a single location to be reviewed independently of other standard functions, like queryUtils, putUtils, and PostUtils.

Reply:
Yeah I can move all of the business logic to a single file, thats a good idea. 

Not sure about passing the property in.  The `applyBusinessLogic`  functions are currently black-boxes that apply any/all business logic to the object/record.  Currently we only have the anonymous logic, but in theory it could be other stuff that is custom to the business.  I could totally see a future enhancement to this, where a config is passed in that tells it which rules to run or skip, and maybe what fields to apply each rule to, etc.  But right now the logic is all the same for each record type, and they all must apply it.

Reply:
Looks good.  Most important I think was to move it into a single location.